### PR TITLE
Feature/refugee profile

### LINKED ISSUE #12: Create refugee profile
Create refugee (family) profile 
(refugee w, w/o his/her family, identification of the camp, medical condition, date of creation)
Note that a family with the family lead member and 0 other family members is also a valid case

### PR DESCRIPTION
**Pull request by michellechan1123**
closes #12

### CODE REVIEW

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:36]: What does this method do? does it rename the refugee?
If so maybe naming it `rename()` would be more clear. 
Also, since the attribute `name` is public now, it doesn't really need a method to change its value. 
If the motivation is for implementing name validation in the future, we should make the `name` attribute private.

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:41]: This docstring doesn't seem to agree with what this method does

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:44]: Why we need to add 1 to the number?

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:49]: Is the method validating the camp? If so, maybe naming it `validate_camp()` would be clearer.

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:58]: What does this method do?

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:70]: Does this mean that the refugee's profiles are still kept after they left the camps?
What is the motivation for keeping them?
I am not sure if we need this, and it may complicate other operations on refugees.

File: models/refugee.py
Comment (dennisylyung) [reply models/refugee.py:49]: but then, I feel like validating camps should actually wait for #3, same for #25 
Therefore I suggest removing this check altogether

**Review by dennisylyung — CHANGES_REQUESTED** [review]

File: models/refugee.py
Comment (dennisylyung) [reply models/refugee.py:78]: Sorry, just noticed that we do need to archive refugees. 
If that is the case, we might need some easy way to check if this refugee is still active

File: models/refugee.py
Comment (Shalaka-Yerawadekar) [inline models/refugee.py:62]: General question: for medical condition, is this the medical condition of the main member of the family or the whole family? If it is the former then having types provided makes sense but if it is the latter then having a description would be better.

File: models/refugee.py
Comment (Shalaka-Yerawadekar) [inline models/refugee.py:94]: Would it be possible to change the message to say:

"Invalid input: the number of family members must be a positive integer."

File: models/refugee.py
Comment (Shalaka-Yerawadekar) [inline models/refugee.py:81]:
```   
 def __str__(self):
        return f"Refugee family {self.name} located in {self.camp}.\n"\
               f"Number of family member: {self.numOfFamilyMember}\n" \
               f"Medical condition: {self.medicalCondition}\n" \
               f"Creation date: {self.dateOfCreation}\n" \
               f"Closing date: {self.dateOfClosing}\n"
               
 ```
 
 Just changing the above so they all start with capital letters.

File: models/refugee.py
Comment (Elynor-Kamil) [inline models/refugee.py:37]: You might also consider checking for whether either firstname or lastname contains a number. Maybe something like:

```
elif firstname.isdigit() or lastname.isdigit():
    raise self.InvalidNameException()
```

Comment (michellechan1123) [reply models/refugee.py:37]: I have added isAlpha() in the elif statement to interrogate if it is a string only with alphabets. 
```
if not isinstance(firstname, str) or not isinstance(lastname, str):
            raise self.InvalidNameException()
        elif not firstname.isalpha() or not lastname.isalpha():
            raise self.InvalidNameException()
```

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:35]: I still prefer the methods to not be named check_xxxx if they return a value. 
Maybe sth like `sanitise_name`?

**Review by dennisylyung — COMMENTED** [review]
Code is much cleaner now! Some comments

**Review by Elynor-Kamil — APPROVED** [review]

**Review by Shalaka-Yerawadekar — APPROVED** [review]

File: models/camp.py
Comment (dennisylyung) [inline models/camp.py:8]: Is this file added to this branch by mistake?
These doesn't look relevant to the issue
(And they doesn't look very right. they should be set from init args right?)

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:51]: Since the default value of `medicalConditionType` is `None`, if the instance is created without a medical condition, the value will become `set(None)`, while what we want is a empty set. 
(e.g. set(None) has len=1, while set() has len=0)
One way to do that is to first check if `medicalConditionType` is `None`, and if so make it an empty set

**Review by dennisylyung — CHANGES_REQUESTED** [review]
Good work! that's a very comprehensive list of medical conditions. 
There are just 2 small issues:
1. Changes in `camp.py` seems to be irrelevant
2. Initialisation without medical conditions seems to be handled incorrectly
Sorry that since they have practical implications I cannot approve for now.

File: models/camp.py
Comment (YunsyYin) [reply models/camp.py:8]: I think the correct file would be:
from models.volunteer import Volunteer

File: models/refugee.py
Comment (YunsyYin) [reply models/refugee.py:93]: I was wondering if it is easier for us to keep the first and last names separately?

**Review by YunsyYin — APPROVED** [review]

File: models/refugee.py
Comment (dennisylyung) [inline models/refugee.py:80]: since you are using emptySet just once, you can just write
```python
else:
    return set()
```
Also, variables should be snake_case

**Review by dennisylyung — APPROVED** [review]
Code looks good. 
Minor code style/non clocking comment

**Review by Shalaka-Yerawadekar — APPROVED** [review]

**Review by Elynor-Kamil — APPROVED** [review]

File: models/refugee.py
Comment (vwcchan) [inline models/refugee.py:33]: Should we change `camp: str` to `camp: Camp` to avoid creating a str object that has no connection with Camp class? Also I think it's be better to have a `InvalidCampException(Exception)` to handle input errors (i.e. when volunteer enter a camp that does not exist).

**Review by vwcchan — APPROVED** [review]
Overall this looks good to me! But I've added one comment and think it may be better if we can adjust that part before merging (also to be align with `Volunteer` class) .

**Review by Elynor-Kamil — APPROVED** [review]

**Review by YingboZhou — APPROVED** [review]

**Pull request merged**
